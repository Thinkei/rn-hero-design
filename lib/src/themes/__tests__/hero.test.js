import heroTheme from '../hero/Hero_Theme.bs';
import legacyTheme from '../legacy/Legacy_Theme.bs';

const getKeys = obj => {
  const keys = Object.keys(obj);
  keys.sort();
  return keys;
};

describe('Theme Hero', () => {
  it('has correct configs', () => {
    expect(heroTheme).toMatchSnapshot();
  });

  it('has exact keys with legacy theme', () => {
    const heroComponents = getKeys(heroTheme);
    const legacyComponents = getKeys(legacyTheme);

    expect(heroComponents).toEqual(legacyComponents);

    heroComponents.forEach(component => {
      expect(getKeys(heroTheme[component])).toEqual(
        getKeys(legacyTheme[component]),
      );
    });
  });
});
