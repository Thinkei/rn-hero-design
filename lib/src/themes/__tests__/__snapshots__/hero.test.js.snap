// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Hero has correct configs 1`] = `
Object {
  "_switch": Object {
    "thumb": Object {
      "backgroundColor": "#fcfcfc",
      "borderRadius": 12.1875,
      "height": 24.375,
      "position": "absolute",
      "top": 0,
      "width": 24.375,
    },
    "thumbOff": Object {
      "left": 0,
    },
    "thumbOn": Object {
      "left": 24.375,
    },
    "thumbWrapper": Object {
      "borderRadius": 12.1875,
      "height": 24.375,
      "width": 48.75,
    },
    "track": Object {
      "borderRadius": 14.1875,
      "height": 28.375,
      "padding": 2,
      "width": 52.75,
    },
    "trackOff": Object {
      "backgroundColor": "#d9dbdf",
    },
    "trackOn": Object {
      "backgroundColor": "#1dbeee",
    },
  },
  "avatar": Object {
    "image": Object {
      "left": 0,
      "position": "absolute",
      "top": 0,
    },
    "largeImage": Object {
      "borderRadius": 48.75,
      "height": 97.5,
      "width": 97.5,
    },
    "largeTitle": Object {
      "fontSize": 48.75,
      "fontWeight": "200",
    },
    "largeTitleWrapper": Object {
      "borderRadius": 48.75,
      "borderWidth": 1.5,
      "height": 97.5,
      "width": 97.5,
    },
    "largeWrapper": Object {
      "height": 97.5,
      "width": 97.5,
    },
    "mediumImage": Object {
      "borderRadius": 24.375,
      "height": 48.75,
      "width": 48.75,
    },
    "mediumTitle": Object {
      "fontSize": 24.375,
      "fontWeight": "300",
    },
    "mediumTitleWrapper": Object {
      "borderRadius": 24.375,
      "borderWidth": 1.2,
      "height": 48.75,
      "width": 48.75,
    },
    "mediumWrapper": Object {
      "height": 48.75,
      "width": 48.75,
    },
    "title": Object {
      "color": "#003580",
      "fontFamily": "Proxima Nova",
    },
    "titleWrapper": Object {
      "alignItems": "center",
      "borderColor": "#003580",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
    },
    "wrapper": Object {},
  },
  "bottomButton": Object {
    "button": Object {
      "alignItems": "center",
      "display": "flex",
      "height": 48.75,
      "justifyContent": "center",
      "paddingHorizontal": 16.25,
    },
    "disabledText": Object {
      "color": "#fcfcfc",
    },
    "disabledWrapper": Object {
      "backgroundColor": "#d9dbdf",
    },
    "loadingIndicator": Object {
      "color": "#fcfcfc",
    },
    "text": Object {
      "color": "#fcfcfc",
      "fontFamily": "Proxima Nova",
      "fontSize": 16.25,
      "lineHeight": 16.25,
    },
    "wrapper": Object {
      "backgroundColor": "#1dbeee",
      "borderRadius": 24.375,
      "marginHorizontal": 16.25,
    },
    "wrapperWithHomeBar": Object {
      "marginBottom": 0,
      "paddingBottom": 0,
    },
    "wrapperWithoutHomeBar": Object {
      "marginBottom": 16.25,
      "paddingBottom": 0,
    },
  },
  "button": Object {
    "disabledText": Object {
      "color": "#fcfcfc",
    },
    "disabledWrapper": Object {
      "backgroundColor": "#d9dbdf",
      "borderWidth": 0,
    },
    "filledLoadingIndicator": Object {
      "color": "#fcfcfc",
    },
    "filledText": Object {
      "color": "#fcfcfc",
    },
    "filledWrapper": Object {
      "backgroundColor": "#1dbeee",
    },
    "outlinedLoadingIndicator": Object {
      "color": "#1dbeee",
    },
    "outlinedText": Object {
      "color": "#1dbeee",
      "fontWeight": "500",
    },
    "outlinedWrapper": Object {
      "backgroundColor": "transparent",
      "borderColor": "#1dbeee",
      "borderWidth": 1.2,
    },
    "text": Object {
      "fontFamily": "Proxima Nova",
      "fontSize": 16.25,
      "lineHeight": 16.25,
    },
    "wrapper": Object {
      "alignItems": "center",
      "borderRadius": 24.375,
      "display": "flex",
      "height": 48.75,
      "justifyContent": "center",
      "paddingHorizontal": 16.25,
    },
  },
  "calendar": Object {
    "blurredDayText": Object {
      "color": "#a3a6ac",
    },
    "currentDayText": Object {
      "color": "#1dbeee",
    },
    "day": Object {
      "alignItems": "center",
      "borderBottomColor": "red",
      "flexBasis": "14.285714285714286%",
      "flexDirection": "row",
      "height": 32.5,
      "justifyContent": "center",
    },
    "dayName": Object {
      "color": "#a3a6ac",
      "fontSize": 13.487499999999999,
      "fontWeight": "500",
    },
    "dayNameSunday": Object {
      "color": "#de350b",
    },
    "dayText": Object {
      "color": "#000000",
      "fontFamily": "Proxima Nova",
      "fontSize": 16.25,
      "fontWeight": "400",
      "lineHeight": 16.25,
    },
    "header": Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingVertical": 8.125,
    },
    "headerButton": Object {
      "height": 36.25,
      "justifyContent": "center",
      "minWidth": 36.25,
      "paddingHorizontal": 8.125,
    },
    "headerTitle": Object {
      "fontFamily": "Proxima Nova",
      "fontSize": 16.25,
      "fontWeight": "500",
    },
    "mark": Object {
      "borderRadius": 2,
      "height": 4,
      "marginHorizontal": 1,
      "width": 4,
    },
    "markedDay": Object {
      "bottom": 4.0625,
      "flexDirection": "row",
      "position": "absolute",
    },
    "monthView": Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
    },
    "selectedDay": Object {
      "backgroundColor": "#1dbeee",
      "borderRadius": 16.25,
      "height": 32.5,
      "position": "absolute",
      "width": 32.5,
    },
    "selectedDayText": Object {
      "color": "#fcfcfc",
      "fontWeight": "500",
    },
    "wrapper": Object {},
  },
  "card": Object {
    "card": Object {
      "backgroundColor": "#fcfcfc",
      "borderRadius": 8.125,
      "overflow": "hidden",
    },
  },
  "container": Object {
    "container": Object {
      "padding": 16.25,
    },
    "fluid": Object {
      "flex": 1,
    },
  },
  "dateTimePicker": Object {
    "action": Object {
      "height": 45.9875,
      "justifyContent": "center",
      "paddingHorizontal": 16.25,
    },
    "actionText": Object {
      "color": "#1dbeee",
      "fontSize": 16.25,
    },
    "actions": Object {
      "borderBottomColor": "#d9dbdf",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "flex-end",
    },
    "wrapper": Object {
      "backgroundColor": "#f5f6f8",
    },
  },
  "icon": Object {
    "icon": Object {
      "color": "#000000",
    },
  },
  "listItem": Object {
    "contentWrapper": Object {
      "alignItems": "center",
      "flexDirection": "row",
    },
    "subtitle": Object {
      "color": "#a3a6ac",
      "fontFamily": "Proxima Nova",
      "fontSize": 13.487499999999999,
      "fontWeight": "500",
      "lineHeight": 18.882499999999997,
    },
    "title": Object {
      "color": "#000000",
      "fontFamily": "Proxima Nova",
      "fontSize": 16.25,
      "fontWeight": "500",
      "lineHeight": 22.75,
    },
    "wrapper": Object {
      "alignItems": "center",
      "borderBottomColor": "#d9dbdf",
      "borderBottomWidth": 0.8,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minHeight": 74.1325,
      "paddingHorizontal": 16.25,
      "paddingVertical": 16.25,
    },
  },
  "message": Object {
    "content": Object {
      "fontFamily": "Proxima Nova",
      "fontSize": 13.487499999999999,
      "fontWeight": "500",
      "textAlign": "center",
    },
    "errorContent": Object {
      "color": "#fcfcfc",
    },
    "errorWrapper": Object {
      "backgroundColor": "#de350b",
    },
    "infoContent": Object {
      "color": "#fcfcfc",
    },
    "infoWrapper": Object {
      "backgroundColor": "#1dbeee",
    },
    "successContent": Object {
      "color": "#fcfcfc",
    },
    "successWrapper": Object {
      "backgroundColor": "#01b39c",
    },
    "warningContent": Object {
      "color": "#000000",
    },
    "warningWrapper": Object {
      "backgroundColor": "#fadb14",
    },
    "wrapper": Object {
      "left": 0,
      "paddingHorizontal": 16.25,
      "paddingVertical": 8.125,
      "right": 0,
      "top": 0,
    },
  },
  "select": Object {
    "action": Object {
      "height": 45.9875,
      "justifyContent": "center",
      "paddingHorizontal": 16.25,
    },
    "actionText": Object {
      "color": "#1dbeee",
      "fontSize": 16.25,
    },
    "actions": Object {
      "borderBottomColor": "#d9dbdf",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "flex-end",
    },
    "dialog": Object {
      "backgroundColor": "#fcfcfc",
      "borderRadius": 2,
      "elevation": 10,
      "maxHeight": "70%",
      "width": "80%",
    },
    "optionText": Object {
      "color": "#000000",
      "fontSize": 16.25,
    },
    "optionWrapper": Object {
      "paddingHorizontal": 16.25,
      "paddingVertical": 16.25,
    },
    "overlay": Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "flex": 1,
      "justifyContent": "center",
    },
    "selectedOptionText": Object {
      "color": "#1dbeee",
      "fontWeight": "400",
    },
    "wrapper": Object {
      "backgroundColor": "#f5f6f8",
    },
  },
  "tabBar": Object {
    "content": Object {
      "flex": 1,
    },
    "navigator": Object {
      "alignItems": "stretch",
      "flexDirection": "row",
    },
    "selectedTab": Object {
      "borderBottomColor": "#1dbeee",
    },
    "tab": Object {
      "alignItems": "center",
      "borderBottomColor": "transparent",
      "borderBottomWidth": 3,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 8.125,
    },
    "wrapper": Object {
      "flex": 1,
    },
  },
  "text": Object {
    "h1": Object {
      "fontSize": 32.5,
    },
    "h2": Object {
      "fontSize": 24.375,
    },
    "h3": Object {
      "fontSize": 19.0125,
    },
    "h4": Object {
      "fontSize": 16.25,
    },
    "h5": Object {
      "fontSize": 13.487499999999999,
    },
    "text": Object {
      "color": "#000000",
      "fontFamily": "Proxima Nova",
      "fontSize": 16.25,
    },
  },
  "textInput": Object {
    "activeIcon": Object {
      "color": "#1dbeee",
    },
    "activeLabel": Object {
      "color": "#1dbeee",
    },
    "activeTextInput": Object {
      "borderBottomColor": "#1dbeee",
      "borderBottomWidth": 1,
    },
    "baseTextInput": Object {
      "color": "#000000",
      "flex": 1,
      "fontFamily": "Proxima Nova",
      "fontSize": 16.25,
      "paddingBottom": 16.25,
      "paddingTop": 16.25,
      "textAlignVertical": "center",
    },
    "disabledBaseTextInput": Object {
      "color": "#a3a6ac",
    },
    "errorIcon": Object {
      "color": "#de350b",
    },
    "errorLabel": Object {
      "color": "#de350b",
    },
    "errorMessage": Object {
      "color": "#de350b",
      "fontFamily": "Proxima Nova",
      "fontSize": 13.487499999999999,
      "fontWeight": "500",
      "height": 13.487499999999999,
      "lineHeight": 13.487499999999999,
      "marginTop": 8.125,
    },
    "errorTextInput": Object {
      "borderBottomColor": "#de350b",
    },
    "icon": Object {
      "color": "#a3a6ac",
      "height": 48.75,
    },
    "label": Object {
      "color": "#a3a6ac",
      "fontFamily": "Proxima Nova",
      "fontSize": 13.487499999999999,
      "fontWeight": "500",
      "height": 13.487499999999999,
      "lineHeight": 13.487499999999999,
    },
    "textInput": Object {
      "alignItems": "center",
      "borderBottomColor": "#d9dbdf",
      "borderBottomWidth": 0.8,
      "flexDirection": "row",
    },
    "wrapper": Object {
      "marginBottom": 16.25,
    },
  },
  "variables": Object {
    "BACKGROUND_BLUE_1": "#347ab7",
    "BACKGROUND_BLUE_2": "#b5eeff",
    "BACKGROUND_BLUE_3": "#e6f7ff",
    "BACKGROUND_COLOR": "#fcfcfc",
    "BASE_SIZE": 16.25,
    "BLACK": "#000000",
    "BLUE": "#1dbeee",
    "BORDER_COLOR": "#d9dbdf",
    "DANGER_COLOR": "#de350b",
    "DARK_PRIMARY_COLOR": "#003580",
    "DISABLED_BACKGROUND_COLOR": "#d9dbdf",
    "DISABLED_TEXT_COLOR": "#a3a6ac",
    "ERROR_COLOR": "#de350b",
    "FOCUS_BLUE_1": "#003580",
    "FOCUS_BLUE_2": "#40679f",
    "FOCUS_BLUE_3": "#5d7ead",
    "FONT_FAMILY": "Proxima Nova",
    "FONT_SIZE": 16.25,
    "GREEN": "#01b39c",
    "GREY_1": "#2b2b2b",
    "GREY_2": "#71767c",
    "GREY_3": "#a3a6ac",
    "GREY_4": "#d9dbdf",
    "GREY_5": "#f5f6f8",
    "HEADER_1": 32.5,
    "HEADER_2": 24.375,
    "HEADER_3": 19.0125,
    "HEADER_4": 16.25,
    "HEADER_5": 13.487499999999999,
    "INFO_COLOR": "#1dbeee",
    "INVERTED_TEXT_COLOR": "#fcfcfc",
    "LARGE_SIZE": 24.375,
    "LINK_TEXT_COLOR": "#1dbeee",
    "MAGENTA": "#ee1d84",
    "MEDIUM_SIZE": 16.25,
    "MUTED_TEXT_COLOR": "#a3a6ac",
    "ORANGE": "#ffa234",
    "PRIMARY_COLOR": "#1dbeee",
    "RED": "#de350b",
    "SMALL_SIZE": 8.125,
    "SUCCESS_COLOR": "#01b39c",
    "TEXT_COLOR": "#000000",
    "WARNING_COLOR": "#fadb14",
    "WHITE": "#fcfcfc",
    "YELLOW": "#fadb14",
  },
}
`;
